Intro to Storybook Spanish translation (React)

lmac-1 opened the pull request:
As discussed in [this issue](https://github.com/chromaui/learnstorybook.com/issues/3), I've been working on an updated Spanish translation of the Intro to Storybook tutorial for React over the past month or so. This PR contains a full translation to all screens of the current English Intro to Storybook tutorial. 

I have also gone through my updates with my Colombian partner to make sure that it makes sense for him.

Let me know if you have any further questions, or need any further updates. 

Thanks!
Lucy

jonniebigodes commented on content/intro-to-storybook/react/es/screen.md:
```diff
@@ -11,146 +11,410 @@ En este capítulo continuaremos aumentando la sofisticación combinando componen
   />
 </video>
 
+### Automatizar pruebas con el corredor de pruebas
```
@lmac-1, one small item here, if you could update the heading and the text below to reference "test runner" instead of "corridor de pruebas" I'd appreciate it. The reasoning behind this is that, in this case, we're referring to a package and not functionality, and the same rules apply as with, let's say, for example, GitHub, React, or Storybook, as they shouldn't be translated. I know this may look a bit weird and kinda breaks the wording by introducing a foreign word in the middle of the text with the choice of naming and for that, I'm sorry for it

lmac-1 replied:
That makes perfect sense. I'll make the update.

lmac-1 replied:
Fixed in [this commit](https://github.com/lmac-1/learnstorybook.com/commit/ea88b2160618bdcd89cb2815ce37cefd9501ae76) ✅

jonniebigodes reviewed (changes requested):
@lmac-1, this looks amazing 🤩 , I scanned through it, and it just needs a couple of tweaks and should be good to go on my end to merge. Let me know once you've made the changes and we'll go from there. Sounds good?

lmac-1 commented on content/intro-to-storybook/react/es/composite-component.md:
```diff
@@ -25,79 +25,87 @@ Comienza con una implementación aproximada de la `TaskList`. Necesitarás impor
+const Template = args => <TaskList {...args} />;
+export const Default = Template.bind({});
+Default.args = {
+  // Dar forma a las historias a través de la composición de args.
+  // Los datos se heredaron de la historia predeterminada en Task.stories.js.
```
No problem, will get on it! Perhaps we should add something about this to the original issue that tracks translations (if it's not there already!)

lmac-1 replied:
Fixed in [this commit](https://github.com/lmac-1/learnstorybook.com/commit/ac11b3972be16abb0c79abf730e5d4a3d9212e7f)  ✅

I made sure all code snippets are equal to the English version, which is why you'll see some extra line breaks added in this commit.

jonniebigodes commented:
@lmac-1 no need to thank. Glad to help you out in proofreading it. Regarding your question about the repository, can you provide me with a bit more context on what you're experiencing? I'm asking because this repository, in comparison with other ones in the Chromatic org or the Storybook org, is one of the smallest ones, and you should be able to clone it without any issues. Related to this, did you, by any chance, try and fork this repository and tried to clone it locally into your machine?

If you didn't, can you please try and let me know? Also, while we're at it, you might need to grab the documentation folder from the Storybook monorepo, which can be done by running the command `yarn extract-sb-docs-metadata` after the cloning process is finished and you installed the required dependencies via running `yarn`. If it doesn't work, please let me know so I can look at it and see what's happening. Sounds good?


Looking forward to hearing from you.

Hope you have a great weekend.

lmac-1 commented:
Previously when I was trying to `git clone` my fork, my terminal was hanging and getting stuck and wasn't able to finish cloning. Today I've managed to clone the project locally, so I can start working on the final updates 🎉

However, I'm not able to test the changes locally because when I start the development server, it's returning the following JSON in the browser: 

`{"codeFrame":"No codeFrame could be generated","sourcePosition":null,"sourceContent":null}`

When I opened the browser, this is the error that is happening in the terminal: 
```
warn The path "/" errored during SSR.
Edit its component webpack/bootstrap:19:1 to resolve the error.

 ERROR #11615 

There was an error while trying to load dev-404-page:
 Cannot find module '../../../../public/page-data/sq/d/1604654434.json'

File: webpack/bootstrap:19:1
```

Also higher up in my terminal, there were some other errors after running `yarn dev`: 

```
ERROR #85923  GRAPHQL

There was an error in your GraphQL query:

Cannot query field "dxData" on type "Query".

If you don't expect "dxData" to exist on the type "Query" it is most
likely a typo. However, if you expect "dxData" to exist there are a
couple of solutions to common problems:

- If you added a new data source and/or changed something inside
gatsby-node/gatsby-config, please try a restart of your development
server.
- You want to optionally use your field "dxData" and right now it is not
used anywhere.

It is recommended to explicitly type your GraphQL schema if you want to
use optional fields.

File: src/components/composite/AppLayout.js:136:7

See our docs page for more info on this error:
https://gatsby.dev/creating-type-definitions


 ERROR #85923  GRAPHQL

There was an error in your GraphQL query:

Cannot query field "dxData" on type "Query".

If you don't expect "dxData" to exist on the type "Query" it is most
likely a typo. However, if you expect "dxData" to exist there are a
couple of solutions to common problems:

- If you added a new data source and/or changed something inside
gatsby-node/gatsby-config, please try a restart of your development
server.
- You want to optionally use your field "dxData" and right now it is not
used anywhere.

It is recommended to explicitly type your GraphQL schema if you want to
use optional fields.

File: src/components/screens/NotFoundScreen/NotFoundScreen.js:22:7

See our docs page for more info on this error:
https://gatsby.dev/creating-type-definitions

failed extract queries from components - 1.457s
success write out requires - 0.062s
success run static queries - 0.112s - 1/1 8.91/s
success run page queries - 0.035s - 3/3 86.03/s
warn Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly:
https://github.com/browserslist/update-db#readme

 ERROR #98124  WEBPACK

Generating development SSR bundle failed

Can't resolve '../../../public/page-data/sq/d/238248339.json' in '/Users/
lucymacartney/Documents/Code/learnstorybook.com/src/components/composite'

If you're trying to use a package make sure that
'../../../public/page-data/sq/d/238248339.json' is installed. If you're
trying to use a local file make sure that the path is correct.

File: src/components/composite/AppLayout.js:3:0


 ERROR #98124  WEBPACK

Generating development SSR bundle failed

Can't resolve '../../../../public/page-data/sq/d/1604654434.json' in
'/Users/lucymacartney/Documents/Code/learnstorybook.com/src/components/sc
reens/NotFoundScreen'

If you're trying to use a package make sure that
'../../../../public/page-data/sq/d/1604654434.json' is installed. If
you're trying to use a local file make sure that the path is correct.

File: src/components/screens/NotFoundScreen/NotFoundScreen.js:1:0

⠀
```


To clarify, these are the steps that previously followed: 
1. `git clone` my fork
2. change branch to `spanish-translation-updates`
3. update version in gatsby-config
4. `yarn install`
5. `yarn extract-sb-docs-metadata`
6. `yarn dev` 

Let me know if I'm missing a step, or configured something slightly incorrectly

lmac-1 commented:
I've made the updates you mentioned: 
- Updated all code snippets to match English version
- Updated Gatsby config
- Removed reference to corredor de prueba
- Updated README to show that this guide's translation is updated

Would I also be able to add myself as a contributor? 😁 Is that in the overall README file, or is it just in `/content/intro-to-storybook/index.md`?

jonniebigodes commented:
@lmac-1, thanks for taking the time to address the feedback. I hope you don't mind, but I've reached out to a fellow team member to take a look at this, as she's more fluent in Spanish than I am, and wanted to see if we missed anything regarding the translation. She's currently checking it, and we'll follow up with you based on any potential feedback. With that out of the way, regarding your issue in previewing the documentation, this is something that we're aware of and will address after Storybook 7.0, as we currently don't have the bandwidth to address it. We already pinpointed it, and it's just a matter of addressing the required changes. For contributors like yourself, I will add you later once this is out. I'm planning on updating the issue, calling out for translators to place you in the right slot. Also, the Readme (repository, not the tutorial) will be updated to feature you as a contributor, similar to other recent contributors who helped us. Once the changes are up, I'll let you know. Sounds good?

lmac-1 commented:
Sounds great! No worries about running locally, completely understand. As long as it's all working on your side, that's great! 

And yes sounds good about contributors. I noticed that some had links to Twitter, in case you need it it's https://twitter.com/codingwithlucy

Thanks, will keep an eye out for any other feedback! :)

lmac-1 commented:
Hi Jonny, hope things are well with you! I just wanted to check in to ask how the translation review is going?  Let me know if you need any other changes done :) 

Thanks and have a great day!

jonniebigodes commented on content/intro-to-storybook/react/es/using-addons.md:
```diff
@@ -0,0 +1,129 @@
+  />
+</video>
+
+Si estamos haciendo [pruebas visuales](/intro-to-storybook/react/es/test/), las pruebas nos dirán si la solución de truncamiento se rompe. Los casos extremos oscuros pueden olvidarse sin la cobertura de prueba.
```
@lmac-1, when you're able, could you update this sentence to:
```md
Si estamos haciendo [pruebas visuales] (/intro-to-storybook/react/es/test/), las pruebas nos dirán si la solución de truncamiento se rompe. Los casos extremos podrían ser ignorados sin la cobertura de prueba.
```
Per my conversation with my fellow team member, that is way more fluent than me in Spanish, the choice of `oscuros` is not incorrect not may have the wrong connotation. She also pointed out that the original English version should include the same change.

lmac-1 replied:
Fixed in commit 665ee7c 🙏🏼

jonniebigodes commented on content/intro-to-storybook/react/es/using-addons.md:
```diff
@@ -0,0 +1,129 @@
+
+Hay complementos para cada posible caso de uso, y demoraría una eternidad en escribir sobre todos. Vamos a integrar uno de los complementos más populares: [Controls](https://storybook.js.org/docs/react/essentials/controls).
+
+## Que es Controls?
```
@lmac-1 not much of a blocker but a nice to have for accuracy and consistency. If you could take another pass at the translation and include the `¿` for questions and it's equivalent to exclamation marks, I'd appreciate it.

lmac-1 replied:
Fixed in commit 665ee7c 🙏🏼

jonniebigodes reviewed (changes requested):
@lmac-1 sorry for the delay and radio silence, but it's been a couple of busy weeks getting stuff ready for 7.0. I've checked in with my fellow team member, and she only had one item that requires attention and one other nice to have so that we're consistent and accurate. Asides from those nothing stood out, and once they're addressed this should be good to be merged. 

Let me know once you've applied the changes, and we'll go from there.

Hope you have a fantastic weekend.

Stay safe

lmac-1 commented:
Completely understand that these things take time @jonniebigodes, especially when other projects are involved. 

I've fixed both of the issues mentioned, so I think we should be good to go. Please thank your colleague for taking the time to review my translations, much appreciated! 

Have a great weekend :-)
